src/dynamic_programming/zero_matrix.md

```diff
@@ -0,0 +1,86 @@
+<!--?title Finding the largest zero submatrix -->
+
+# Finding the largest zero submatrix
+
+You are given a matrix with `n` rows and `m` columns. Find the largest submatrix consisting of only zeros (a submatrix is a rectangular area of the matrix).
+
+## Algorithm
+
+Elements of the matrix will be `a[i][j]`, where `i = 0...n - 1`, `j = 0... m - 1`. For simplicity, we will consider all non-zero elements equal to 1.
+
+### Step 1: Auxiliary dynamic
+
+First, we calculate the following auxiliary matrix: `d[i][j]`, nearest row that has a 1 above `a[i][j]`. Formally speaking, `d[i][j]` is the largest row number (from `1` to `i`), in which there is a element equal to `1` in the `j`-th column. 
+While iterating from top-left to bottom-right, when we stand in row `i`, we know the values from the previous row, so, it is enough to update just the elements with value `1`. We can save the values in a simple array `d[i]`, `i = 1...m - 1`:
+
+```cpp
+vector<int> d (m, -1);
+for (int i=0; i<n; ++i) {
+  for (int j=0; j<m; ++j){
+    if (a[i][j] == 1){
+      d[j] = i;
+    }
+  }
+}
+```
+
+### Step 2: Problem solving
+
+We can solve the problem in `O(nm^2)` iterating through rows, considering every possible left and right columns for a submatrix. The bottom of the rectangle will be the current row, and using `d[i][j]` we can find the top row. However, it is possible to go further and significantly improve the complexity of the solution.
+
+It is clear that the desired zero submatrix is bounded on all four sides by some ones, which prevent it from increasing in size and improving the answer. Therefore,  we will not miss the answer if we act as follows: first, we iterate over the `i` bottom row of the zero submatrix, then we iterate over which column `j` we will rest up on the zero submatrix. Using the value `d[i][j]`, we immediately get the number of the top line of the zero submatrix. It now remains to determine the optimal left and right boundaries of the zero submatrix, i.e. maximally push this submatrix to the left and right of the `j`-th column. 
```

Comment:
Maybe reword the first part a little bit. It is quite hard to understand, what exactly you are doing.

Reply:
I mean the part how you span up the submatrix using the j-th cell in the i-th row, go up to d[i][j].
Maybe something like that: for every cell j in in a row i (the bottom row of a potential zero submatrix), I go up...